### ISSUE TITLE
Estimator API - Unexpected behaviour for Validation Loss during early training steps

### ISSUE THREAD

**Issue by michaeltrs** [tensorflow/tensorflow#29495]
**System information**
- Have I written custom code : Yes
- OS Platform and Distribution : Linux Ubuntu 18.04.2 LTS
- TensorFlow installed from : binary
- TensorFlow version : 1.12.0
- Python version: 2.7.15
- CUDA/cuDNN version: 10.1/7
- GPU model and memory: GeForce GTX 1080Ti 11Gb

I am trying the TF Estimator API for monitoring validation metrics during training. To debug my code I have set up some runs with the following:

- train and val sets are the same, both consisting of 1 sample
- batch size is set to 1 both for training and validation
- learning rate is set to 0.0 so that no changes happen to the network

**Describe the expected behavior**
I would expect training and validation loss to be constant throughout training and equal to each other

**Describe the current behavior**
However, what is observed in Tensorboard is:
- training loss behaves as expected
- validation loss is different in the beginning, slowly converging towards the constant value of the training loss
- increasing the learn rate seems to reduce this effect significantly. In the plots below all is the same apart from the learn rates which are 0.001 and 0.1

My question is if this is a bug or this behaviour is expected and if so what is the underlying mechanism driving this? 

lear_rate = 0.0
![image](https://user-images.githubusercontent.com/18511714/59041124-ea96a780-886f-11e9-94d5-1e3db786902e.png)

learn_rate = 0.001
![image](https://user-images.githubusercontent.com/18511714/59041183-0b5efd00-8870-11e9-9061-bff53435292a.png)

learn_rate = 0.1
![image](https://user-images.githubusercontent.com/18511714/59041217-174abf00-8870-11e9-8ad2-eed13161450e.png)

**Code to reproduce the issue**
```
import tensorflow as tf

data_paths = "/home/user/Desktop/test_csv_files/1sample.csv"
batch_size = 1

def get_batch(paths, options):
    ....
    return {"images": image_features_tensor}, labels_tensor
    
def fully_connected_network(features, sizes, is_training):
    inputs = features['images']
    if sizes == [0]:
        return inputs
    for i, size in enumerate(sizes):
        inputs = tf.layers.dense(inputs=inputs, units=size)
        inputs = tf.nn.relu(inputs)
        inputs = tf.layers.batch_normalization(
            inputs, momentum=0.99, axis=-1, epsilon=0.001, training=is_training, 
            reuse=None, fused=USE_FUSED_BN)
    return inputs
    
def model_fn(features, labels, mode, params):
    if mode == tf.estimator.ModeKeys.TRAIN:
        is_training = True
    else:
        is_training = False
    print("IS TRAINING: ", is_training)

    features = fully_connected_network(features, 1, is_training)
    logits = tf.layers.dense(inputs=features, units=3)
    softmax = tf.nn.softmax(logits, name="softmax_out")
    predicted_classes = tf.reshape(tf.math.argmax(softmax, axis=1, output_type=tf.int32), (-1, 1))

    predictions = {"classes": predicted_classes, "probabilities": softmax}

    if mode == tf.estimator.ModeKeys.PREDICT:
        return tf.estimator.EstimatorSpec(mode=mode, predictions=predictions)

    one_hot_labels = tf.one_hot(labels, depth=3)
    loss = tf.reduce_mean(
        tf.nn.softmax_cross_entropy_with_logits(labels=one_hot_labels, logits=logits))

    batch_accuracy = batch_multiclass_metrics(labels, predicted_classes)

    tf.summary.scalar('accuracy', batch_accuracy)

    if mode == tf.estimator.ModeKeys.TRAIN:
        initial_learn_rate = tf.constant(learn_rate, tf.float32)
        learn_rate = tf.train.exponential_decay(learning_rate=0.001,
            global_step=tf.train.get_global_step(), decay_steps=1000,
            decay_rate=0.95,staircase=True)
        optimizer = tf.train.AdamOptimizer(learning_rate=learn_rate)
        update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
        train_op = optimizer.minimize(
            loss=loss, global_step=tf.train.get_global_step())
        train_op = tf.group([train_op, update_ops])
        return tf.estimator.EstimatorSpec(
            mode=mode, loss=loss, train_op=train_op)  # ,
    if mode == tf.estimator.ModeKeys.EVAL:
        eval_metric_ops = {
            "accuracy": tf.metrics.accuracy(labels=labels, 
                                            predictions=predicted_classes)}
        return tf.estimator.EstimatorSpec(mode=mode,
                                          loss=loss,
                                          predictions=predictions,  # test
                                          eval_metric_ops=eval_metric_ops)
                                          
checkpoint_config = tf.estimator.RunConfig(
        save_checkpoints_steps=1000,
        save_summary_steps=200,
        keep_checkpoint_max=10)

estimator = tf.estimator.Estimator(
    model_fn=model_fn,
    params=config_model_options,
    model_dir=savedir,
    config=checkpoint_config,
    warm_start_from=None)

train_input_fn = lambda paths: get_batch(data_paths, options=config_train_options)
eval_input_fn = lambda paths: get_batch(data_paths, options=config_val_options)

train_spec = tf.estimator.TrainSpec(input_fn=lambda: train_input_fn(data__paths),
                                    max_steps=None)

eval_spec = tf.estimator.EvalSpec(input_fn=lambda: eval_input_fn(data_paths),
                                  steps=1,
                                  start_delay_secs=20,
                                  throttle_secs=20)

tf.estimator.train_and_evaluate(estimator=estimator,
                                train_spec=train_spec,
                                eval_spec=eval_spec)  
```

**Other info / logs**
The question was originally asked in stack overflow https://stackoverflow.com/questions/56188656/tensorflow-estimator-api-validation-loss-during-early-training-steps

Many thanks,
Michael

**Comment by gadagashwini** [comment]
@michaeltrs Looks input sample.csv is missing to reproduce the reported issue here. Please provide the input data. Thanks!

**Comment by gadagashwini** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
